### ISSUE TITLE
Send a message to all my friends will block by account?

### ISSUE THREAD

**Issue by miguelwon** [carpedm20/fbchat#427]
Hi, 
This is not really an issue with `fbchat`, just a question I would like to know from your experience working with `fbchat`. I want to send a message to all my friends (a few hundred).  Do you know if there is a good possibility that my account will be blocked by facebook, due to spam?

I was thinking of send perhaps the message to 20 friends every 10 seconds.

**Comment by MikSDigital** [comment]
As far as I know FB is very strict with mass messages. So chance is high your activity can be seen as suspicious and may be some approval will be asked from FB that you are human.

**Comment by tcurdt** [comment]
I am investigating the same thing @miguelwon is trying to do. I am not really concerned about being classified as spam as it the number of messages/time will be very low. But I am wondering how to even get a list of all friends - which does read very differently than what `fetchAllUsers` does.
Is that even possible with *fbchat*?

**Comment by miguelwon** [comment]
Update on this: I have tried to send a message at a rate of one each 10s and after around 70 messages, FB "blocked" me, i.e., it requested an "I'm not a robot" proof before sending a new message. 

About the `fetchAllUsers`. It returns only the friends of which you exchanged a message in the past. Is not all friends but near because each time a new friendship is accepted a small automatic message is sent through messenger.

**Comment by tcurdt** [comment]
@miguelwon Aren't those automatic messages a "newer" thing? So very old contacts won't be included? At least I don't think it was like that from the beginning. Would be nice to reach all.

**Comment by madsmtm** [comment]
@tcurdt, see https://github.com/carpedm20/fbchat/issues/282, I'm not really sure, but I don't actually think we ended up fixing the issue 😕

**Comment by tcurdt** [comment]
That's a shame - but thanks for the pointer. Thanks, @madsmtm.

**Comment by tcurdt** [comment]
I just tried and I was just listing my friends. I called the script 3 times and my account got temporarily suspended (eeek!) After changing the password fb was happy again. They detected "unusual activity". I guess this raises the question how to avoid this.

For one I forgot to logout in the script - which may or may not be a problem. On the other hand I assume that too frequent logins are more a problem than missing logouts.
I also noticed that the reported browser was different for all detected logins.

Maybe it would make sense to set the browser agent to something fix and "unsuspicious" and retain a single session? Any thoughts?

**Comment by miguelwon** [comment]
Are you using cookies sessions? If not that may be the reason. I had a similar issue because I was trying to log in many different sessions in a short period of time. You should log in once, save the session and the log in again but with the same cookies: 

```
session_cookies = client.getSession()
client = Client('<email>', '<password>', session_cookies=session_cookies)
```

Even with this facebook complains if the login is made from a location it never saw before. For this case, after you set the new password and check it's you that is trying to log in, you should wait a few hours to log in again from that remote location.

**Comment by tcurdt** [comment]
@miguelwon I guess that's what I am after then. Thanks.
I used the code from the examples - which probably should be adjusted then.

**Comment by tcurdt** [comment]
From a logical point of view - shouldn't something like this be enough then?

```
client = Client(session_cookies=session_cookies)
```

If there is already the login stored in the cookies - why the username and password?

**Comment by madsmtm** [comment]
@tcurdt, because if the session cookies are invalid / outdated, the client will attempt to login instead 😉

**Comment by adbenitez** [comment]
> Even with this facebook complains if the login is made from a location it never saw before. For this case, after you set the new password and check it's you that is trying to log in, you should wait a few hours to log in again from that remote location.

this is my situation, I want to setup an email-messenger bridge, but the bot is in a remote server in another country, @madsmtm there are a way to confirm it is me with fbchat? like sending the verification code, since I can't manually access facebook from the remote server,
Thanks in advance

**Comment by kapi2289** [comment]
@adbenitez I've used my remote server as VPN by SSH (with PuTTy) and from my computer logged into Facebook with this VPN. Weird, but works.

**Comment by adbenitez** [comment]
@kapi2289, yes tis is really weird, since I asked to a friend in Spain to create a test account and I was able to login here in my country with that account, but with other 2 test accounts created in my country and trying to login from Spain it ask for verification  by confirming from other logged device or by identifying friends pictures.
This works similar to 2FA so maybe it can be implemented in fbchat, the url is also m.facebook.com/checkpoint

**Comment by adbenitez** [comment]
@kapi2289 also, you logged in with your browser with cookies/session already salved? then it makes sense that it works for you

**Comment by kapi2289** [comment]
I've used my remote server's VPN to create and log into the account, and then I could use this account on this remote server. Or previously I was installing desktop browser on the remote server and I was doing it from there, but that's not the best idea :sweat_smile:

**Comment by adbenitez** [comment]
well, an easy workaround I had found for this "location/device changed" issue is to enable 2FA for the accounts
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.
